### DISCUSSION TITLE
DataType rules and completion

### DISCUSSION THREAD

**Discussion by cdietrich** [eclipse-langium/langium#1599] [Q&A]
Hi, is it intentional, that for datatype rules, the completionFor is not called with a rulecall for the complete datatype rule but its parts only

**Comment by msujew** [comment]
Hi @cdietrich,

the completion provider is supposed to provide full completion for a data type rule see [this code](https://github.com/eclipse-langium/langium/blob/3b007c2be9a6ec0c93f4dac62eea33854909db60/packages/langium/src/lsp/completion/completion-provider.ts#L238-L251) and [these tests](https://github.com/eclipse-langium/langium/blob/3b007c2be9a6ec0c93f4dac62eea33854909db60/packages/langium/test/lsp/completion-provider.test.ts#L308-L397). What kind of behavior are you seeing?

**Reply by cdietrich** [reply]
given grammar

```
entry Model:
    (definitions+=Definition)*
    ;

Definition:
    'define' name=QualifiedName;


ID returns string: RawId | EscapedId;
QualifiedName returns string: ID QNF*;
fragment QNF returns string: ('.' ID);

hidden terminal WS: /\s+/;
terminal RawId: /[_a-zA-Z][\w_-]*/;
terminal EscapedId: /`[^`]*`/;
```

and logging

```
protected override completionFor(context: CompletionContext, next: NextFeature, acceptor: CompletionAcceptor): MaybePromise<void> {
        if (GrammarAST.isRuleCall(next.feature)) {
            console.log(next.feature.rule.ref?.name)
        }  
        return super.completionFor(context, next, acceptor)
      }
```

on model

```
define |
```

it will log

```
RawId
EscapedId
```

and not ID or QualifedName

=> no change to find out i am completing the name in Define
